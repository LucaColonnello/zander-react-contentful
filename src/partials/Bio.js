import React, { Component } from 'react'
import { getEntry } from '../utils/contentful'
import ReactMarkdown from 'react-markdown'

class Home extends Component {
  state = {
    bio: ''
  }

  componentDidMount () {
    console.log('Fetching bio entry...')
    if (window.localStorage.getItem('bio')) {
      this.setState({bio: JSON.parse(window.localStorage.getItem('bio'))})
      return;
    }
    getEntry('3AgxXXOhNCkqQEYkESSwCU')
      .then((response) => {
        this.setState({bio: response.fields.bio})
        window.localStorage.setItem('bio', JSON.stringify(response.fields.bio))
        console.log(response.fields);
      })
      .catch((error) => {
        console.log('error occured')
        console.log(error)
      })
  }

  render () {
    return(
      <div>
        <ReactMarkdown source={this.state.bio}/>
      </div>
    )
  }
}

export default Home
