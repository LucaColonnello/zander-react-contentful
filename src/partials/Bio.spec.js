import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Bio from './Bio';

test('Bio snapshot test', () => {
  const component = shallow(<Bio />);
  const tree = shallowToJson(component);
  expect(tree).toMatchSnapshot();
});
