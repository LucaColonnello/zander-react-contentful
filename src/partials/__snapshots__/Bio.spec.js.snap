// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bio snapshot test 1`] = `
<div>
  <ReactMarkdown
    containerTagName="div"
    parserOptions={Object {}}
    source=""
  />
</div>
`;
